sgl-kernel/CMakeLists.txt

```diff
@@ -112,8 +110,9 @@ option(SGL_KERNEL_ENABLE_SM90A  "Enable SM90A"  OFF)
 option(SGL_KERNEL_ENABLE_BF16   "Enable BF16"   ON)
 option(SGL_KERNEL_ENABLE_FP8    "Enable FP8"    ON)
 option(SGL_KERNEL_ENABLE_FP4    "Enable FP4"    OFF)
+option(BUILD_FA3   "Enable FA3"   OFF)
 
-if ("${CUDA_VERSION}" VERSION_GREATER_EQUAL "12.8" OR SGL_KERNEL_ENABLE_SM100A)
+if ("${CUDA_VERSION}" VERSION_GREATER_EQUAL "12.8" AND SGL_KERNEL_ENABLE_SM100A)
```

Comment:
I think the `OR` logic is correct. When building, we only need to consider the CUDA version. If it is equal to or greater than 12.8, we should include the sm100a feature. Alternatively, if the user specifies sm100a, we should also enable this feature as intended.

Reply:
The building machine can be a GPU or CPU node. Please refer to this link: https://github.com/sgl-project/sglang/issues/5062#issue-2971599473